src/symmetry/crystal_symmetry.cpp

```diff
@@ -214,26 +220,32 @@ Unit_cell_symmetry::Unit_cell_symmetry(matrix3d<double> const& lattice_vectors__
         if (spg_dataset_->n_atoms != num_atoms__) {
             std::stringstream s;
             s << "spg_get_dataset() returned wrong number of atoms (" << spg_dataset_->n_atoms << ")" << std::endl
-              << "expected number of atoms is " <<  num_atoms__;
+              << "expected number of atoms is " << num_atoms__;
             RTE_THROW(s);
         }
     }
-    PROFILE_STOP("sirius::Unit_cell_symmetry|spg");
+    PROFILE_STOP("sirius::Crystal_symmetry|spg");
 
     if (spg_dataset_) {
+        auto lat_sym = find_lat_sym(lattice_vectors_, tolerance_);
         /* make a list of crystal symmetries */
         for (int isym = 0; isym < spg_dataset_->n_operations; isym++) {
-            auto sym_op = get_spg_sym_op(isym, spg_dataset_, lattice_vectors__, num_atoms__, positions__, tolerance__);
-            /* add symmetry operation to a list */
-            space_group_symmetry_.push_back(sym_op);
+            matrix3d<int> R(spg_dataset_->rotations[isym]);
+            for (auto& e: lat_sym) {
+                if (e == R) {
```

Comment:
That's the key point of this PR: we accept only part of SPGLIB symmetries for which the lattice is invariant under rotation with a given tolerance